tests/ranged_balance.cpp

```diff
@@ -334,8 +337,19 @@ static void range_test( const std::array<double, 5> &test_thresholds )
             WARN( "No matching dispersion found" );
         } else {
             WARN( "Range: " << r << " Dispersion: " << found_dispersion );
+            data.push_back( found_dispersion );
         }
     }
+    if( write_data )
+    {
+        std::ofstream out_file;
```

Comment:
Please consider using `write_to_file` instead of using `ofstream` directly.

Reply:
I'm getting a really weird error after implementing write_to_file. Did I do something wrong?

`.Terminated: data/mods//dda/../../json/hit_range.json: line 1:1: member not foun
d: type

{
^
"even_good":[1723,859,573,429,343,283,245,214,190,171,156,142,132,122,114,107,10
1,88,88,85,81,78,74,71,68,66,63,61,59,57,55,53,52,50,49,47,46,45,44,42,41,41,39,
39,38,37,36,35,35,34,33,33,32,31,31,30,30,29,29,28]}}}}}}}}}}}}}}}}}}}}}}}}}}}}}`